inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/cache/RealTimeQuery.java

```diff
@@ -0,0 +1,312 @@
+
+    private final List<DataSource> dataSourceList = new LinkedList<>();
+
+    private final String queryLogTsSql;
+    private final String queryIdsByIpSql;
+    private final String queryReportIpsSql;
+
+    private RealTimeQuery() {
+        List<JdbcConfig> jdbcConfigList = ConfigService.getInstance().getAllAuditSource();
+        for (JdbcConfig jdbcConfig : jdbcConfigList) {
+            dataSourceList.add(createDataSource(jdbcConfig));
+        }
+
+        queryLogTsSql = Configuration.getInstance().get(KEY_SOURCE_QUERY_MINUTE_SQL,
+                DEFAULT_SOURCE_QUERY_MINUTE_SQL);
+        queryIdsByIpSql = Configuration.getInstance().get(KEY_SOURCE_QUERY_IDS_SQL,
+                DEFAULT_SOURCE_QUERY_IDS_SQL);
+        queryReportIpsSql = Configuration.getInstance().get(KEY_SOURCE_QUERY_IPS_SQL,
+                DEFAULT_SOURCE_QUERY_IPS_SQL);
+    }
+
+    public static RealTimeQuery getInstance() {
+        if (realTimeQuery == null) {
+            synchronized (Configuration.class) {
+                if (realTimeQuery == null) {
+                    realTimeQuery = new RealTimeQuery();
+                }
+            }
+        }
+        return realTimeQuery;
+    }
+
+    /**
+     * Create data source.
+     */
+    private DataSource createDataSource(JdbcConfig jdbcConfig) {
+        HikariConfig config = new HikariConfig();
+        config.setDriverClassName(jdbcConfig.getDriverClass());
+        config.setJdbcUrl(jdbcConfig.getJdbcUrl());
+        config.setUsername(jdbcConfig.getUserName());
+        config.setPassword(jdbcConfig.getPassword());
+        config.setConnectionTimeout(Configuration.getInstance().get(KEY_DATASOURCE_CONNECTION_TIMEOUT,
+                DEFAULT_CONNECTION_TIMEOUT));
+        config.addDataSourceProperty(CACHE_PREP_STMTS,
+                Configuration.getInstance().get(KEY_CACHE_PREP_STMTS, DEFAULT_CACHE_PREP_STMTS));
+        config.addDataSourceProperty(PREP_STMT_CACHE_SIZE,
+                Configuration.getInstance().get(KEY_PREP_STMT_CACHE_SIZE, DEFAULT_PREP_STMT_CACHE_SIZE));
+        config.addDataSourceProperty(PREP_STMT_CACHE_SQL_LIMIT,
+                Configuration.getInstance().get(KEY_PREP_STMT_CACHE_SQL_LIMIT, DEFAULT_PREP_STMT_CACHE_SQL_LIMIT));
+        config.setMaximumPoolSize(
+                Configuration.getInstance().get(KEY_DATASOURCE_POOL_SIZE,
+                        DEFAULT_DATASOURCE_POOL_SIZE));
+        return new HikariDataSource(config);
+    }
+
+    /**
+     * Query the audit data of log time.
+     *
+     * @param startTime
+     * @param endTime
+     * @param inlongGroupId
+     * @param inlongStreamId
+     * @param auditId
+     * @return
+     */
+    public List<StatData> queryLogTs(String startTime, String endTime, String inlongGroupId,
+            String inlongStreamId, String auditId) {
+        List<StatData> statDataList = new LinkedList<>();
+        for (DataSource dataSource : dataSourceList) {
```

Comment:
Random access to the data sources to ensure the load balance of each data source.

Reply:
There is no need to check every audit data source here, and the weight of the previous data sources will be greater.